test/tests.py

```diff
@@ -0,0 +1,161 @@
+from operations import create_index, check_index, create_document,\
+    delete_document, delete_index, get_document
+from http import HTTPStatus
+from typing import Tuple, Callable
+import unittest
+import os
+import logging
+import time
+import requests
+from requests.exceptions import ConnectionError, SSLError
+
+logger = logging.getLogger(__name__)
+
+
+# The following "retry_request" function's purpose is to retry a certain request for "max_attempts"
+# times every "delay" seconds IF the requests returned a status code other than what's expected.
+# So this "retry_request" function's arguments are a request function's name and whatever arguments that function
+# expects, and the status code the request function is expecting to get.
+def retry_request(request: Callable, args: Tuple = (), max_attempts: int = 10, delay: float = 0.5,
+                  expected_status_code: HTTPStatus = None):
+    for attempt in range(1, max_attempts + 1):
+        try:
+            result = request(*args)
+            if result.status_code == expected_status_code:
+                return result
+            else:
+                logger.warning(f"Status code returned: {result.status_code} did not"
+                               f" match the expected status code: {expected_status_code}."
+                               f" Trying again in {delay} seconds.")
+                time.sleep(delay)
+        except ConnectionError as e:
+            logger.error(f"Received exception: {e}. Unable to connect to server. Please check all containers are up"
+                         f" and ports are setup properly")
+            logger.warning(f"Trying again in {delay} seconds.")
+            time.sleep(delay)
+            continue
+        except SSLError as e:
+            logger.error(f"Received exception: {e}. Unable to connect to server. Please check all containers are up"
+                         f"and ports are setup properly")
+            logger.warning(f"Trying again in {delay} seconds.")
+            time.sleep(delay)
+            continue
+    logger.error(f"All {max_attempts} attempts failed.")
+    logger.error(f"Couldn't get the expected status code: {expected_status_code} while making the request:"
+                 f"{request.__name__} using the following arguments: {args} ")
+    return None
```

Comment:
If you are going to return None rather than raise the exception you should check for this case. Otherwise your tests below can fail with an attribute error.

Reply:
Thanks for pointing this out. I didn't think it was an issue since if the tests do fail with an attribute error, the logs would mention that an exception had happened and would show the retry attempts, but now just realized that it this will give more clarity to the user, and clearer test results. Just pushed a commit to address that.